### PR TITLE
Add `staticModifiers` option

### LINKED ISSUE #603: Add `staticModifiers` flag (to support shaking unused modifiers and route splitting)
`AppFiles` does not handle modifiers, they are all put under `otherAppFiles`. I believe they should be treated similar to helpers, so they get split into route bundles, right?

### PR DESCRIPTION
**Pull request by Windvis**
~~still need to add tests and actually test if it works :grimacing:~~ I added some tests. Let me know if I missed something :pray:.

Closes #603

### CODE REVIEW

File: packages/compat/src/resolver-transform.ts [hunk 1]
```diff
@@ -111,6 +111,16 @@ export function makeResolverTransform(resolver: Resolver) {
             }
           }
         },
+        ElementModifierStatement(node: ASTv1.ElementModifierStatement) {
+          if (node.path.type !== 'PathExpression') {
+            return;
+          }
+          if (node.path.this === true) {
```
Comment (lifeart) [inline packages/compat/src/resolver-transform.ts:118]: we need same for `node.path.data === true`  ({{@fooBar}})

Comment (Windvis) [reply packages/compat/src/resolver-transform.ts:118]: Thanks, didn't know about that!

Comment (Windvis) [reply packages/compat/src/resolver-transform.ts:118]: Added! I also forgot about "contextual" modifiers so I added that as well.

Comment (lifeart) [reply packages/compat/src/resolver-transform.ts:121]: @Windvis could we cover "modifier" helper usage?

like

```hbs
{{let (modifier "foo-bar" arg param=1) as |my-modifier|}}
   <div {{my-modifier}} ></div>
{{/let}}
```

**Review by lifeart — APPROVED** [review]

**Comment by Windvis** [comment]
CI seems to fail due to #1024 which should be fixed by #1025

**Review by lifeart — APPROVED** [review]

**Pull request merged**